File: AI-agent-backend-fastapi/app/plugins/api_engine/schemas/execution.py
# Copyright (c) 2025 左岚. All rights reserved.
"""
执行记录Schema
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime


class ExecutionResponse(BaseModel):
    """执行记录响应Schema"""
    execution_id: int = Field(..., description="执行记录ID")
    case_id: int = Field(..., description="用例ID")
    task_id: str = Field(..., description="Celery任务ID")
    status: str = Field(..., description="执行状态")
    result: Optional[Dict[str, Any]] = Field(None, description="执行结果")
    logs: Optional[str] = Field(None, description="执行日志")
    error_message: Optional[str] = Field(None, description="错误信息")
    duration: Optional[float] = Field(None, description="执行时长(秒)")
    steps_total: int = Field(0, description="总步骤数")
    steps_passed: int = Field(0, description="通过步骤数")
    steps_failed: int = Field(0, description="失败步骤数")
    executed_by: int = Field(..., description="执行人ID")
    executed_at: datetime = Field(..., description="执行时间")
    finished_at: Optional[datetime] = Field(None, description="完成时间")
    
    class Config:
        from_attributes = True


class ExecutionListResponse(BaseModel):
    """执行记录列表响应Schema"""
    total: int = Field(..., description="总数")
    items: List[ExecutionResponse] = Field(..., description="执行记录列表")


class ExecutionStatusResponse(BaseModel):
    """执行状态响应Schema"""
    task_id: str = Field(..., description="任务ID")
    status: str = Field(..., description="任务状态")
    progress: Optional[int] = Field(None, ge=0, le=100, description="进度百分比")
    current_step: Optional[str] = Field(None, description="当前步骤")
    message: Optional[str] = Field(None, description="状态消息")
    result: Optional[Dict[str, Any]] = Field(None, description="执行结果")

File: AI-agent-frontend-vue3/src/plugins/api-engine/api/execution.ts
/**
 * API引擎插件 - 执行管理API
 */
import request from '@/utils/request'

export interface Execution {
    execution_id?: number
    case_id: number
    task_id?: string // Celery任务ID
    status: string // pending/running/success/failed/error
    result?: any // 执行结果(JSON格式)
    logs?: string // 执行日志
    error_message?: string // 错误信息
    duration?: number // 执行时长(秒)
    steps_total?: number // 总步骤数
    steps_passed?: number // 通过步骤数
    steps_failed?: number // 失败步骤数
    executed_by?: number // 执行人ID
    executed_at?: string // 执行时间
    finished_at?: string // 完成时间
}

export interface ExecutionListParams {
    page?: number
    page_size?: number
    case_id?: number
    status?: string
}

export const executionAPI = {
    /**
     * 执行测试用例
     */
    executeCase(caseId: number, context?: any) {
        return request.post(`/api/v1/plugin/api-engine/executions/${caseId}/execute`, {
            context: context || {}
        })
    },

    /**
     * 查询执行状态
     */
    getExecutionStatus(taskId: string) {
        return request.get(`/api/v1/plugin/api-engine/executions/task/${taskId}/status`)
    },

    /**
     * 获取执行历史列表
     */
    getExecutions(params?: ExecutionListParams) {
        return request.get('/api/v1/plugin/api-engine/executions', { params })
    },

    /**
     * 获取执行详情
     */
    getExecutionById(id: number) {
        return request.get(`/api/v1/plugin/api-engine/executions/${id}`)
    },

    /**
     * 删除执行记录
     */
    deleteExecution(id: number) {
        return request.delete(`/api/v1/plugin/api-engine/executions/${id}`)
    },

    /**
     * SSE日志流URL (需要在组件中使用EventSource)
     */
    getLogStreamUrl(taskId: string): string {
        const baseURL = import.meta.env.VITE_API_BASE_URL || 'http://localhost:8000'
        return `${baseURL}/api/v1/plugin/api-engine/executions/task/${taskId}/log-stream`
    }
}

